src/backend/utils/text-parser.js

```diff
@@ -1,20 +1,28 @@
-const puppeteer = require('puppeteer');
+const { Cluster } = require('puppeteer-cluster');
 
-module.exports = async function(htmlFragment) {
-  let browser;
-  try {
-    browser = await puppeteer.launch();
-    const page = await browser.newPage();
+let isInitialized = false;
+let cluster;
 
-    await page.setContent(htmlFragment, {
-      waitUntil: 'domcontentloaded',
+module.exports.parse = async function(htmlFragment) {
+  if (!isInitialized) {
+    cluster = await Cluster.launch({
+      concurrency: Cluster.CONCURRENCY_CONTEXT,
```

Comment:
Should be safe to use `CONCURRENCY_PAGE`

Reply:
will be fixed soon